### PR TITLE
telegram-downloader 0.12.1 (new formula).

### PR DESCRIPTION
**Pull request by astrosheep-zero**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Please stop making new PRs after every comment, it makes it very hard to review them when you do not have the history.

https://github.com/Homebrew/homebrew-core/pull/150528 or https://github.com/Homebrew/homebrew-core/pull/153887 should have stayed open instead of making this one

**Comment by thekoc** [comment]
> Please stop making new PRs after every comment, it makes it very hard to review them when you do not have the history.
> 
> #150528 or #153887 should have stayed open instead of making this one

Sorry about that. I just intended to make the commit history simple.

**Comment by chenrui333** [comment]
> > Please stop making new PRs after every comment, it makes it very hard to review them when you do not have the history.
> > #150528 or #153887 should have stayed open instead of making this one
> 
> Sorry about that. I just intended to make the commit history simple.

you should actually branching out (ie not use `master` so that maintainers can help with the forced PR update :) ) otherwise, you can always squash and rebase to make clean history

worst case, you can just re-commit the change in the a fresh new commit and forced update your PR.

File: Formula/t/telegram-downloader.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  desc "Telegram Messenger downloader/tools written in Golang"
+  homepage "https://github.com/iyear/tdl"
+  url "https://github.com/iyear/tdl/archive/refs/tags/v0.12.1.tar.gz"
+  sha256 "09229d333db613ce55155125b04f1e3269113586dca10b5e24d47450fc73f4ae"
+  license "AGPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(output: bin/"tdl", ldflags: "-s -w")
```
Comment (chenrui333) [inline Formula/t/telegram-downloader.rb:11]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w", output: bin/"tdl")
```

File: Formula/t/telegram-downloader.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    assert_match "# ID of dialog", shell_output("#{bin}/tdl chat ls -f -").strip
+    assert_match "callback: not authorized. please login first", shell_output("#{bin}/tdl chat ls -n _test", 1).strip
```
Comment (chenrui333) [inline Formula/t/telegram-downloader.rb:16]: this should just work?

```suggestion
    assert_match "# ID of dialog", shell_output("#{bin}/tdl chat ls -f -")
    assert_match "callback: not authorized. please login first", shell_output("#{bin}/tdl chat ls -n _test", 1)
```

**Review by chenrui333 — COMMENTED** [review]
LGTM (just two nit comments)

when you include the new changes, remember to squash the changes and forced update your branch. :)

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@thekoc thanks a lot for your first contribution to homebrew-core! 🎉

**Pull request merged**